libsoftwarecontainer/src/container.cpp

```diff
@@ -110,11 +110,16 @@ bool Container::initialize()
 {
     if (m_state < ContainerState::PREPARED) {
         std::string gatewayDir = gatewaysDir();
-        if (!createDirectory(gatewayDir)) {
+        if (!m_create.createDirectory(gatewayDir)) {
+            m_create.rollBack();
             log_error() << "Could not create gateway directory "
                         << gatewayDir << ": " << strerror(errno);
             return false;
         }
+        m_create.clear();
+        if (!pathInList(gatewayDir)) {
+            m_cleanupHandlers.emplace_back(new DirectoryCleanUpHandler(gatewayDir));
+        }
```

Comment:
Here is another example of how this change creates double checks instead of using the existing FileToolkitWithUndo.